sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs

```diff
@@ -48,15 +48,25 @@ public static HttpPipeline Build(ClientOptions options, bool bufferResponse = tr
         // internal for testing
         internal static TelemetryPolicy CreateTelemetryPolicy(ClientOptions options)
         {
+            const string PackagePrefix = "Azure.";
+
             Assembly clientAssembly = options.GetType().Assembly;
-            AzureSdkClientLibraryAttribute componentAttribute = clientAssembly.GetCustomAttribute<AzureSdkClientLibraryAttribute>();
-            if (componentAttribute == null)
+
+            AssemblyInformationalVersionAttribute versionAttribute = clientAssembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>();
+            if (versionAttribute == null)
+            {
+                throw new InvalidOperationException("Client SDK assemblies are required to have InformationalVersionAttribute");
```

Comment:
Can we make this error message a little more helpful?  The old one included the client assembly name, but I'd love it to also say something along the lines of

`$"{nameof(AssemblyInformationalVersionAttribute)} is required on client SDK assembly '{clientAssembly.FullName}' (inferred from the use of options type '{options.GetType().FullName}')."`

as a hint to partner teams adding new client libraries.  I tried using the default `ClientOptions` type when I was first building a library and had to jump down the telemetry rabbit hole in the debugger.  😄